File: truth_set/python/logic/performance/emailsToFind.py
users = [
    {"id": i + 1, "email": f"user{i + 1}@example.com"}
    for i in range(100000)
]

emails_to_find = ['user100@example.com', 'user200@example.com', 'user300@example.com']

for email in emails_to_find:
    user = next((u for u in users if u['email'] == email), None)
    if user:
        print(user['id'])
\ No newline at end of file

File: truth_set/go/logic/performance/emailsToFind.go
package performance

import (
    "fmt"
)

type User struct {
    ID    int
    Email string
}

func emailsToFind() {
    users := make([]User, 100000)
    for i := range users {
        users[i] = User{
            ID:    i + 1,
            Email: fmt.Sprintf("user%d@example.com", i+1),
        }
    }

    emailsToFind := []string{"user100@example.com", "user200@example.com", "user300@example.com"}

    for _, email := range emailsToFind {
        for _, user := range users {
            if user.Email == email {
                fmt.Println(user.ID)
                break
            }
        }
    }
}
\ No newline at end of file